File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxco = 0;
        for(int i=0;i<nums.size();i++){
            if(i > maxco){
                return false;
            }
            maxco = max(maxco, i+nums[i]);
        }
        return true;
    }
   
};

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxcount = 0;
        for(int i=0;i<nums.length;i++){
            if(i > maxcount){
                return false;
            }
            maxcount = Math.max(maxcount, i + nums[i]);
            
        }
        return true;
        
    }
}